buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -642,6 +642,10 @@ private void runElaticsearchBinScript(String tool, String... args) {
         defaultEnv.put("ES_TMPDIR", tmpDir.toString());
         // Windows requires this as it defaults to `c:\windows` despite ES_TMPDIR
         defaultEnv.put("TMP", tmpDir.toString());
+        // Windows needs the COMPUTERNAME variable in order to set HOSTNAME
+        if (System.getenv("COMPUTERNAME") != null) {
```

Comment:
Where is HOSTNAME passed through for linux? Do we actually need to rely on the real system computername or could we pass in a dummy one for tests?

Reply:
HOSTNAME is a builtin for Bash, so we don't need to pass it through. I believe it's always available unless you explicitly unset it:

https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html

I think a dummy value will work and avoid the need for a null check.

Reply:
In fact, the integration test that was failing here just checks that the node name is not an empty string. I'm going to pass in HOSTNAME and COMPUTERNAME explicitly and test for specific values.